### ISSUE TITLE
2.3.4 Fatal error if WooCommerce Admin not active

### ISSUE THREAD

**Issue by seldimi** [woocommerce/facebook-for-woocommerce#1849]
When updating to 2.3.4 and you don’t have WooCommerce Admin installed, you get a fatal error:

PHP Fatal error: Uncaught Error: Class ‘Automattic\WooCommerce\Admin\Features\Features’ not found in /home/www/…./wp-content/plugins/facebook-for-woocommerce/includes/Admin/Settings.php:62

**Comment by iMazed** [comment]
This was also reported in 3869417-hc, and can be replicated on a fresh install. Setting to critical.

**Comment by iMazed** [comment]
Also reported on WP.org: https://wordpress.org/support/topic/fatal-error-if-woocommerce-admin-not-active/

**Comment by jedkirby** [comment]
WooCommerce Admin is packaged up within WooCommerce directly, so, this issue is present because the version 2.3.4 relies on at least WooCommerce 5.0.0 > being installed.

WooCommerce introduced the class `Automattic\WooCommerce\Admin\Features\Features` after version 5.0.0, as see below:

WooCommerce v4.9.2: https://plugins.trac.wordpress.org/browser/woocommerce/tags/4.9.2/packages/woocommerce-admin/src/Features
WooCommerce v5.0.0: https://plugins.trac.wordpress.org/browser/woocommerce/tags/5.0.0/packages/woocommerce-admin/src/Features?order=name

To resolve your instance, you'll need to either downgrade Facebook for WooCommerce, or upgrade WooCommerce to 5.0.0 or greater.

**Comment by seldimi** [comment]
Well, that doesn't make sense. There are a lot of websites using WC 4.x or need some more time to be upgraded to 5.x version. 
Also, there was neither notice, nor minimum requirements about depreciating WC 4.x version. 

Lastly, have a look at the plugin requirements
WC requires at least: 3.5.0

You should be more prepared if you want to discontinue WC 4.x

**Comment by jedkirby** [comment]
I totally agree, it's a breaking change. I don't work on this plugin, FYI - i'm just resolving broken WooCommerce website's as well as you.. Hopefully they'll resolve the issue soon, but, in the mean time downgrading as I mentioned above fixes it for the short term.

**Comment by iMazed** [comment]
@jedkirby - the website I'm working on is running WooCommerce 5.1.0 so I'm pretty sure that there's an additional problem going on. Thanks for your input, though!

**Comment by jedkirby** [comment]
@iMazed - only thing I could think of is that the separate **woocommerce-admin** package is installed on your setup, on version `1.8.3` or less; from that tag upwards, the `Features` class was introduced, see: https://plugins.trac.wordpress.org/browser/woocommerce-admin/tags/1.8.3/src/Features. If not, then i'm out of ideas .. hopefully enough all together for the developers to take a look into.

**Comment by budzanowski** [comment]
We should have a patch very soon, thank you for creating the issue.

**Comment by rcstr** [comment]
Hi @seldimi - thank you for reporting this issue, We have release version `2.4.5` which fixes this issue. The release can be manually downloaded via https://github.com/woocommerce/facebook-for-woocommerce/releases/download/2.3.5/facebook-for-woocommerce.2.3.5.zip

cc @iMazed

**Comment by adrianpaun** [comment]
> Hi @seldimi - thank you for reporting this issue, We have release version `2.4.5` which fixes this issue. The release can be manually downloaded via https://github.com/woocommerce/facebook-for-woocommerce/releases/download/2.3.5/facebook-for-woocommerce.2.3.5.zip
> 
> cc @iMazed

Worked for me, on WC 4.8.0, Thanks for this quick fix.
